pkg/obfuscate/appsec.go

```diff
@@ -0,0 +1,538 @@
+			case scanBeginLiteral:
+				// Let the following stepToOneOf do the job and scan until the next array value
+				err = nil
+			}
+			if err != nil {
+				return i
+			}
+		}
+		// Step to the beginning of the next array value or end of the array
+		var op int
+		i, op, err = stepToOneOf(scanner, input, i, scanArrayValue, scanEndArray)
+		if err != nil || op == scanEndArray {
+			return i
+		}
+	}
+	return i
+}
+
+// Obfuscate the parameter object of the form `{ "key_path": <key path>, "highlight": <highlight>, "value": <value>, <any extra keys>: <any extra values> }`.
+// The implementation is permissive so that any extra keys and values are allowed, and permits having any of the three
+// keys `key_path`, `highlight` and `value` that we need to obfuscate.
+// Note that the overall parameter obfuscation directly depends on the presence of a sensitive key in the key_path.
+// As a result, the parameter object needs to be entirely walked to firstly find the key_path.
+func (o *appsecEventsObfuscator) obfuscateRuleMatchParameter(scanner *scanner, input string, i int, diff *inputDiff) (int, error) {
+	// Walk the object and save the `key_path` value along with the offsets of the
+	// `highlight` and `value` values, if any.
+	var (
+		paramKeyPath                         string
+		paramValueFrom, paramValueTo         int
+		paramHighlightFrom, paramHighlightTo int
+	)
+	i, err := walkObject(scanner, input, i, func(keyFrom, keyTo int, valueFrom, valueTo int) {
+		switch input[keyFrom:keyTo] {
+		case `"key_path"`:
+			paramKeyPath = input[valueFrom:valueTo]
+		case `"value"`:
+			paramValueFrom = valueFrom
+			paramValueTo = valueTo
+		case `"highlight"`:
+			paramHighlightFrom = valueFrom
+			paramHighlightTo = valueTo
+		}
+	})
+	if err != nil {
+		return i, err
+	}
+	// Firstly start by searching for any sensitive key into the key_path
+	var hasSensitiveKey bool
+	if paramKeyPath != "" {
+		hasSensitiveKey = o.hasSensitiveKeyPath(paramKeyPath)
+	}
+	// Finally, obfuscate the `highlight` and `value` values
+	if highlights := input[paramHighlightFrom:paramHighlightTo]; highlights != "" {
+		var tmpDiff inputDiff
+		o.obfuscateRuleMatchParameterHighlights(highlights, &tmpDiff, hasSensitiveKey)
+		tmpDiff.appendTo(diff, paramHighlightFrom)
+	}
+	if value := input[paramValueFrom:paramValueTo]; value != "" {
+		var tmpDiff inputDiff
+		o.obfuscateRuleMatchParameterValue(value, &tmpDiff, hasSensitiveKey)
+		tmpDiff.appendTo(diff, paramValueFrom)
+	}
+	return i, nil
+}
+
+// Return whether the given key path contains any sensitive key. A key is
+// considered sensitive when the key regular expression matches it. It only
+// applies to key path elements of string type.
+// The expected key path value is of the form `[ <path 1>, <...>, <path N> ]`.
+// The implementation is permissive so that any array value type is accepted.
+func (o *appsecEventsObfuscator) hasSensitiveKeyPath(keyPath string) (hasSensitiveKey bool) {
```

Comment:
```suggestion
func (o *appsecEventsObfuscator) hasSensitiveKeyPath(keyPath string) bool {
```
I think named return values only make sense when the return value is ambiguous, for the sake of documentation, or in situations where there are many return values (which is a bit similar in reasoning).

Reply:
Fixed by https://github.com/DataDog/datadog-agent/pull/11324/commits/581d45615b0733a482044bf79caec4c43e21be83